scripts/scil_compute_local_tracking.py

```diff
@@ -219,17 +221,20 @@ def tracks_generator_wrapper():
             if args.save_seeds:
                 dps['seeds'] = seed
 
-            # IMPORTANT: Streamlines are dumped in world space (mm) with
-            # origin `corner`.
-            strl += 0.5
-            strl *= voxel_size  # in mm.
+            # TODO: Use nibabel utilities for dealing with spaces
+            if ext == '.trk':
+                # Streamlines are dumped in mm space with
+                # origin `corner`.
+                strl += 0.5
+                strl *= voxel_size  # in mm.
+            else:
+                # True world space with origin center
+                strl = np.dot(strl, odf_sh_img.affine[:3, :3]) +\
+                    odf_sh_img.affine[:3, 3]
             if args.compress:
                 strl = compress_streamlines(strl, args.compress)
             yield TractogramItem(strl, dps, {})
 
-    # TODO: LazyTractogram.from_data_func seems to expect streamlines in
-    # world coordinates (mm) with origin `corner` although this is not
-    # specified anywhere
     tractogram = LazyTractogram.from_data_func(tracks_generator_wrapper)
     tractogram.affine_to_rasmm = odf_sh_img.affine
```

Comment:
Here is how I call it somewhere else in my code, does it work if you call it like that?

def streamlines_func(): return (e for e in streamlines)
obj = LazyTractogram(streamlines_func
                     affine_to_rasmm=affine)

Reply:
I understand, but won't data all be stored in RAM  before `streamlines_func()` returns. Shouldn't the streamlines func yield items? Also I think I had a problem with dumping my seeds on-the-fly too.